### PR TITLE
Support migration to existing spanner databases

### PR DESCRIPTION
**Pull request by Deep1998**
Added a fix for the schema-only failure.

### CODE REVIEW

File: conversion/conversion.go [hunk 1]
```diff
@@ -417,12 +417,85 @@ func getSeekable(f *os.File) (*os.File, int64, error) {
...
+		return dbExists, err
+	}
+	if dbExists {
+		err = ValidateDDL(ctx, adminClient, dbURI)
+	}
+	return dbExists, err
+}
+
+// CheckExistingDb checks whether the dbURI exists or not.
+func CheckExistingDb(ctx context.Context, adminClient *database.DatabaseAdminClient, dbURI string) (bool, error) {
```
Comment (bharadwaj-aditya) [inline conversion/conversion.go:440]: make the methods private where it is not expected to be called from outside

File: conversion/conversion.go [hunk 2]
```diff
@@ -444,17 +517,42 @@ func CreateDatabase(project, instance, dbName string, conv *internal.Conv, out *
...
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return "", fmt.Errorf("can't create admin client: %w", analyzeError(err, project, instance))
+	}
+	defer adminClient.Close()
+
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, dbName)
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database:   dbURI,
+		Statements: conv.SpSchema.GetDDL(ddl.Config{Comments: false, ProtectIds: true, Tables: true, ForeignKeys: false}),
```
Comment (bharadwaj-aditya) [inline conversion/conversion.go:537]: why is ForeignKeys false ? Is this handled elsewhere ?

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]

File: conversion/conversion.go (hunk 1 above)
Comment (Deep1998) [reply conversion/conversion.go:440]: These are being called in the unit tests thats why kept it public.

File: conversion/conversion.go (hunk 2 above)
Comment (Deep1998) [reply conversion/conversion.go:537]: Yes foreign keys we create at the very end after data migration. This only creates tables and indexes.

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

File: conversion/conversion.go (hunk 2 above)
Comment (agasheesh) [reply conversion/conversion.go:537]: I think it's fair to add comment to the same effect to help future readers understand this.

**Comment by agasheesh** [comment]
The title and description for this PR is pretty thin, which makes it hard for future readers to navigate the changes. In general, there should be enough detail in title for readers to "find" the appropriate PR and in description for readers to not to have to look at the code to understand the changes, unless they want to.

File: conversion/conversion.go [hunk 3]
```diff
@@ -417,12 +417,85 @@ func getSeekable(f *os.File) (*os.File, int64, error) {
...
+	if err != nil {
+		return dbExists, err
+	}
+	if dbExists {
+		err = ValidateDDL(ctx, adminClient, dbURI)
+	}
+	return dbExists, err
+}
+
+// CheckExistingDb checks whether the dbURI exists or not.
```
Comment (agasheesh) [inline conversion/conversion.go:439]: `whether the dbURI` -> `whether database with dbURI`

**Review by agasheesh — APPROVED** [review]

**Pull request merged**